src/foundry/common/types.d.mts

```diff
@@ -390,26 +411,26 @@ declare global {
     /**
      * An error, if one occurred
      */
-    error?: Error;
+    error?: Error | undefined;
 
     /**
      * The status of the request
      */
-    status?: string;
+    status?: string | undefined;
 
     /**
      * The ID of the requesting User
      */
-    userId?: string;
+    userId?: string | undefined;
 
     /**
      * Data returned as a result of the request
      */
-    data?: RequestData;
+    data?: RequestData | undefined;
 
     /**
      * An Array of created data objects
      */
-    result?: object[] | string[];
+    result?: object[] | string[] | undefined;
```

Comment:
1. Note that this has been offloaded to another type, RequestData
2. Replace `object` with `Record<string, unknown>`